flight/Modules/OnScreenDisplay/inc/osd_utils.h

```diff
@@ -0,0 +1,177 @@
+	if (mode == 0) { stroke = 0; fill = 1; } \
+	if (mode == 1) { stroke = 1; fill = 0; } \
+
+// Line endcaps (for horizontal and vertical lines.)
+#define ENDCAP_NONE  0
+#define ENDCAP_ROUND 1
+#define ENDCAP_FLAT  2
+
+#define DRAW_ENDCAP_HLINE(e, x, y, s, f, l) \
+	if ((e) == ENDCAP_ROUND) /* single pixel endcap */ \
+{ write_pixel_lm(x, y, f, l); } \
+	else if ((e) == ENDCAP_FLAT) /* flat endcap: FIXME, quicker to draw a vertical line(?) */ \
+{ write_pixel_lm(x, y - 1, s, l); write_pixel_lm(x, y, s, l); write_pixel_lm(x, y + 1, s, l); }
+
+#define DRAW_ENDCAP_VLINE(e, x, y, s, f, l) \
+	if ((e) == ENDCAP_ROUND) /* single pixel endcap */ \
+{ write_pixel_lm(x, y, f, l); } \
+	else if ((e) == ENDCAP_FLAT) /* flat endcap: FIXME, quicker to draw a horizontal line(?) */ \
+{ write_pixel_lm(x - 1, y, s, l); write_pixel_lm(x, y, s, l); write_pixel_lm(x + 1, y, s, l); }
+
+// Macros for writing pixels in a midpoint circle algorithm.
+#define CIRCLE_PLOT_8(buff, cx, cy, x, y, mode) \
+	CIRCLE_PLOT_4(buff, cx, cy, x, y, mode); \
+	if ((x) != (y)) { CIRCLE_PLOT_4(buff, cx, cy, y, x, mode); }
+
+#define CIRCLE_PLOT_4(buff, cx, cy, x, y, mode) \
+	write_pixel(buff, (cx) + (x), (cy) + (y), mode); \
+	write_pixel(buff, (cx) - (x), (cy) + (y), mode); \
+	write_pixel(buff, (cx) + (x), (cy) - (y), mode); \
+	write_pixel(buff, (cx) - (x), (cy) - (y), mode);
+
+// Font flags.
+#define FONT_BOLD      1               // bold text (no outline)
+#define FONT_INVERT    2               // invert: border white, inside black
+// Text alignments.
+#define TEXT_VA_TOP    0
+#define TEXT_VA_MIDDLE 1
+#define TEXT_VA_BOTTOM 2
+#define TEXT_HA_LEFT   0
+#define TEXT_HA_CENTER 1
+#define TEXT_HA_RIGHT  2
+
+// Max/Min macros
+#define MAX3(a, b, c)                MAX(a, MAX(b, c))
+#define MIN3(a, b, c)                MIN(a, MIN(b, c))
+#define LIMIT(x, l, h)               MAX(l, MIN(x, h))
+
+// Check if coordinates are valid. If not, return. Assumes signed coordinates for working correct also with values lesser than 0.
+#define CHECK_COORDS(x, y)           if (x < GRAPHICS_LEFT || x > GRAPHICS_RIGHT || y < GRAPHICS_TOP || y > GRAPHICS_BOTTOM) { return; }
+#define CHECK_COORD_X(x)             if (x < GRAPHICS_LEFT || x > GRAPHICS_RIGHT) { return; }
+#define CHECK_COORD_Y(y)             if (y < GRAPHICS_TOP  || y > GRAPHICS_BOTTOM) { return; }
+
+// Clip coordinates out of range. Assumes signed coordinates for working correct also with values lesser than 0.
+#define CLIP_COORDS(x, y)            { CLIP_COORD_X(x); CLIP_COORD_Y(y); }
+#define CLIP_COORD_X(x)              { x = x < GRAPHICS_LEFT ? GRAPHICS_LEFT : x > GRAPHICS_RIGHT ? GRAPHICS_RIGHT : x; }
+#define CLIP_COORD_Y(y)              { y = y < GRAPHICS_TOP ? GRAPHICS_TOP : y > GRAPHICS_BOTTOM ? GRAPHICS_BOTTOM : y; }
+
+// Macro to swap two variables using XOR swap.
+#define SWAP(a, b)                   { a ^= b; b ^= a; a ^= b; }
+
+
+// Text dimension structures.
+struct FontDimensions {
+	int width, height;
```

Comment:
Not that it particularly bothers me, but this would have been better as the more pedantic:
```
struct FontDimensions {
 	int width;
 	int height;
}
```